tests/phpunit/EnvVarProcessorTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DatadirTests\Tests;
+
+use Keboola\DatadirTests\EnvVarProcessor;
+use Keboola\DatadirTests\Exception\EnvVariableNotFoundException;
+use Keboola\DatadirTests\Exception\UnexpectedTypeException;
+use PHPUnit\Framework\Assert;
+use PHPUnit\Framework\TestCase;
+
+class EnvVarProcessorTest extends TestCase
+{
+    /**
+     * @dataProvider getValidInputs
+     * @param mixed $expected
+     */
+    public function testReplaceEnv(?string $var, ?string $value, string $statement, $expected): void
+    {
+        if ($var) {
+            putenv("$var=$value");
+        }
+
+        $processor = new EnvVarProcessor();
+        Assert::assertSame($expected, $processor->replaceEnv($statement));
+    }
+
+    public function getValidInputs(): array
+    {
+        return [
+            'no_var_1' => [null, null, 'abc', 'abc'],
+            'no_var_2' => [null, null, '123', '123'],
+            'string-empty' => ['VAR1', '', '%env(string:VAR1)%', ''],
+            'string-value' => ['VAR1', 'var 1 content', '%env(string:VAR1)%', 'var 1 content'],
+            'int-empty' => ['VAR1', '', '%env(int:VAR1)%', 0],
+            'int-string' => ['VAR1', 'content', '%env(int:VAR1)%', 0],
+            'int-ok' => ['VAR1', '456', '%env(int:VAR1)%', 456],
+            'float-empty' => ['MY_FLOAT_VAR_123', '', '%env(float:MY_FLOAT_VAR_123)%', 0.0],
+            'float-string' => ['MY_FLOAT_VAR_123', 'content', '%env(float:MY_FLOAT_VAR_123)%', 0.0],
+            'float-int' => ['VAR1', '456', '%env(float:VAR1)%', 456.0],
+            'float-ok' => ['VAR1', '456.78', '%env(float:VAR1)%', 456.78],
+            'bool-invalid' => ['VAR1', 'content', '%env(bool:VAR1)%', false],
```

Comment:
tohle bych logicky čekal, že bude true :/

Reply:
ok, otocim to ako pisem vyssie.

Reply:
https://github.com/keboola/datadir-tests/pull/29/commits/2f5bdae1c5f209853af0fbee756821366c5b1e6c